crates/swc_ecma_transforms_module/src/system_js.rs

```diff
@@ -0,0 +1,1143 @@
+                                .clone()
+                                .make_member(quote_ident!(meta.export_names.remove(0))),
+                        )),
+                        right: Box::new(meta.export_values.remove(0)),
+                    }
+                    .into_stmt(),
+                );
+            }
+            meta.setter_fn_stmts.push(
+                CallExpr {
+                    span: DUMMY_SP,
+                    callee: self.export_ident.clone().as_callee(),
+                    args: vec![export_obj.as_arg()],
+                    type_args: None,
+                }
+                .into_stmt(),
+            );
+        }
+
+        meta.setter_fn_stmts
+    }
+
+    fn add_module_item_meta(&mut self, mut m: ModuleItemMeta) {
+        match self
+            .module_item_meta_list
+            .iter()
+            .position(|i| m.src.eq(&i.src))
+        {
+            Some(index) => {
+                let mut meta = self.module_item_meta_list.remove(index);
+                meta.setter_fn_stmts.append(&mut m.setter_fn_stmts);
+                meta.export_names.append(&mut m.export_names);
+                meta.export_values.append(&mut m.export_values);
+                if m.has_export_all {
+                    meta.has_export_all = m.has_export_all;
+                }
+                self.module_item_meta_list.insert(index, meta);
+            }
+            None => {
+                self.module_item_meta_list.push(m);
+            }
+        }
+    }
+
+    fn hoist_var_decl(&mut self, var_decl: VarDecl) -> Option<Expr> {
+        let mut exprs = vec![];
+        for var_declartor in var_decl.decls {
+            let mut tos = vec![];
+            var_declartor.visit_with(&mut VarCollector { to: &mut tos });
+
+            for (sym, ctxt) in tos {
+                if var_declartor.init.is_none() {
+                    for (k, v) in self.export_map.iter_mut() {
+                        if (sym.clone(), ctxt) == *k {
+                            for value in v.iter() {
+                                self.export_names.push(value.clone());
+                                self.export_values.push(*undefined(DUMMY_SP));
+                            }
+                            break;
+                        }
+                    }
+                }
+                self.declare_var_idents
+                    .push(Ident::new(sym, DUMMY_SP.with_ctxt(ctxt)));
+            }
+
+            if let Some(init) = var_declartor.init {
+                exprs.push(Box::new(Expr::Assign(AssignExpr {
+                    span: DUMMY_SP,
+                    op: op!("="),
+                    left: PatOrExpr::Pat(Box::new(var_declartor.name)),
+                    right: init,
+                })));
+            }
+        }
+        match exprs.len() {
+            0 => None,
+            _ => Some(Expr::Seq(SeqExpr {
+                span: DUMMY_SP,
+                exprs: exprs.into_iter().collect(),
```

Comment:
This can be changed to `exprs`.

Reply:
`rustc` will optimize it into `exprs` as core has a specialization for this case, but we should not rely on it.

Reply:
(As it's an impl detail)